File: android/src/main/kotlin/com/github/patou/flutter_view_pdf/FlutterViewPdfPlugin.kt
package com.github.patou.flutter_view_pdf

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** FlutterViewPdfPlugin */
class FlutterViewPdfPlugin: FlutterPlugin {
  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    FlutterPdfViewApi.setUp(flutterPluginBinding.binaryMessenger, FlutterPdfViewApiImpl(flutterPluginBinding))
  }

  override fun onDetachedFromEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    FlutterPdfViewApi.setUp(flutterPluginBinding.binaryMessenger, null)
  }
}

File: ios/Classes/FlutterViewPdfPlugin.swift
import Flutter
import UIKit

public class FlutterViewPdfPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let api = FlutterPdfViewApiImpl()
        FlutterPdfViewApiSetup.setUp(binaryMessenger: registrar.messenger(), api: api)
    }
}
